### PR TITLE
IC4 submission

### PR DESCRIPTION
**Pull request by SandyyMarie**

### CODE REVIEW

File: lib/author.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require './lib/book'
+class Author
+
+  attr_reader :first_name, :last_name, :name, :books, :new_book, :author_first_name, :author_last_name, :title, :publication_date
```
Comment (epintozzi) [inline lib/author.rb:4]: `name` and `books` are the only `attr_readers` you're actually using. The rest can be removed.

File: lib/library.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  def initialize(library_name)
+    @name = library_name
+    @books = []
+    @authors = []
+  end
+
+  def add_author(incoming_author)
+    @authors << incoming_author.name
+    @books << incoming_author.books
```
Comment (epintozzi) [inline lib/library.rb:13]: This should put the whole `Author` object in the array, not just the name so `@authors << incoming_author` is all we need.

For the books, this is close, but the README indicates a flat array containing all books so we should iterate instead. This currently returns an array of arrays, where the nested array is the collection of the individual author's books. Instead we could write this as: 
```ruby
incoming_author.books.each do |book|
  @books << book
end
```
Alternatively, you could add a `flatten!` to the end of line 13 in its current state.

File: lib/library.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  def publication_time_frame_for(given_author)
+    date_range = []
+    given_author.books.each do |book|
+      date_range << book.publication_year
+    end
+    date_range.sort
+    start_end_range = []
+    :start == date_range[0]
+    :end == date.range.last
+    start_end_range = :start, :end
```
Comment (epintozzi) [inline lib/library.rb:25]: I'm not totally sure where you were going with lines 22-25, however up through line 21 looks good! If we add a variable to line 21 so we can use that result, we can finish up this method like this:
```suggestion
    sorted_range = date_range.sort
    { start: sorted_range.first,
      end: sorted_range.last }
```

File: lib/book.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    @author_first_name = input[:author_first_name]
+    @author_last_name = input[:author_last_name]
+    @title = input[:title]
+    @publication_date = input[:publication_date]
+  end
+
+  def publication_year
+    @publication_date
+    temp_array = @publication_date.split
+    year_yanker = temp_array[2]
```
[the remark this answers is not in the document]
Comment (SandyyMarie) [reply lib/book.rb:15]: I really like this fix thank you!

**Pull request closed without merging**